backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/callout/C_Campaign_Price.java

```diff
@@ -70,10 +77,47 @@ public void onBPartnerChanged(final I_C_Campaign_Price record)
 		{
 			return;
 		}
-
+		updatePricingInfo(record);
 		record.setC_BP_Group_ID(-1);
 	}
 
+	private static void updatePricingInfo(final I_C_Campaign_Price record)
+	{
+		final BPartnerId bpartnerId = BPartnerId.ofRepoIdOrNull(record.getC_BPartner_ID());
+		if (bpartnerId == null)
+		{
+			return;
+		}
+		final IBPartnerDAO bpartnerRepo = Services.get(IBPartnerDAO.class);
+		final CountryId countryId = bpartnerRepo.getDefaultShipToLocationCountryIdOrNull(bpartnerId);
+		if (countryId != null)
+		{
+			record.setC_Country_ID(countryId.getRepoId());
+			updateCurrency(record, countryId);
+		}
+		final PricingSystemId pricingSystemId = bpartnerRepo.retrievePricingSystemIdOrNull(bpartnerId, SOTrx.SALES);
+		if (pricingSystemId != null)
+		{
+			record.setM_PricingSystem_ID(pricingSystemId.getRepoId());
+			final IPriceListDAO priceListDAO = new PlainPriceListDAO();
+			final PriceListId priceListId = priceListDAO.retrievePriceListIdByPricingSyst(pricingSystemId, countryId, SOTrx.SALES);
+			if (priceListId != null)
+			{
+				final I_M_PriceList_Version newestPriceListVersion = priceListDAO.retrieveNewestPriceListVersion(priceListId);
+				final ProductId productId = ProductId.ofRepoIdOrNull(record.getM_Product_ID());
+				final I_M_ProductPrice productPrice = ProductPrices.retrieveMainProductPriceOrNull(newestPriceListVersion, productId);
+				if (productPrice != null)
```

Comment:
you shall not retrieve this manually; the price shall be computed using PrincingRUle and from PricingResult you shall take the needed infos.

Reply:
correction done